### ISSUE TITLE
[3.2] Database backup file size doubled 

### ISSUE THREAD

**Issue by ccsantos23** [craftcms/cms#4571]
Is it normal for the database backup file to double in size after upgrading to 3.2?
For example, my last backup for 3.1.33 was 12149 KB zipped, and then these for 3.2:
3.2.0  29842 KB
3.2.2  25420 KB

I'm used to seeing it grow gradually as we entered content, but authoring has mostly settled down so I'm assuming this is from the upgrade to 3.2.

- Craft version: 3.2.2
- PHP version: 7.2.20
- Database driver & version: MySQL 5.5.5
- Plugins & versions:
Element Map	1.2.1
Field Reveal	1.0.1
Fractal	1.0.8
Logs	3.0.0
Matrix Toolbar	1.0.6
Redactor	2.3.3.2
SEOmatic	3.2.14
Smith	1.0.0
Snitch	3.0.0
Workflow	1.2.1

**Comment by brandonkelly** [comment]
Doubling in size isn’t expected, no, especially for 3.2.1+. (3.2.0 attempted to store snapshots of each entry revision, but that was removed in 3.2.1.)

Are you seeing double the entries in the UI?

**Comment by ccsantos23** [comment]
No, I'm not seeing double the entries in the UI.  It has 1343 entries right now, but I didn't note how many there were in 3.1.33.  I did check a couple sections and the number of entries hadn't changed.

[I tested the backup on my dev environment, and that did show a size reduction.  My dev database is small (~1.2 MB) and doesn't have many entries. ]

**Comment by angrybrad** [comment]
Run this query on a 3.1 and 3.2 database backup to see which table file sizes are growing between the two:

```
SELECT 
    table_name AS `Table`, 
    round(((data_length + index_length) / 1024 / 1024), 2) as `Size`
FROM information_schema.TABLES 
WHERE table_schema = "craft3"
order by `Size` DESC
```

**Comment by ccsantos23** [comment]
There was an easier way to see the sizes on phpmyadmin (it was staring right at me on the Structure view, almost missed it).  Attached pdfs of the results.  The tables that grew were: elements-sites, matrixcontent*, content, relations, elements, matrixblocks, entries.  Some of them look like they grew by a factor of 6, which is the number of sites I have.  

Could it be the propagation settings?  I have most of the sections set to propagate to all the sites.  And I did re-propagate (by unchecking and then checking the propagate box), in preparation for our coming translated content.  But that was back when we were still on 3.1.*.

[tmp-craftdb-rows-3-1-33.pdf](https://github.com/craftcms/cms/files/3395334/tmp-craftdb-rows-3-1-33.pdf)
[tmp-craftdb-rows-3-2-2.pdf](https://github.com/craftcms/cms/files/3395335/tmp-craftdb-rows-3-2-2.pdf)

**Comment by ccsantos23** [comment]
I Looked closer at one of the simpler tables: matrixcontent_homebannermatrix, which had 30 rows in 3.1.33, and then 2088 rows in 3.2.2.  I went in to edit the page that had that matrix, checked the list of revisions, then got out.  Before the page changed, I saw that it created a draft even though I didn't make any changes.  
Then I checked the row count for the table, and it had gone up to 2154 rows.
I went back into edit mode to check: there were 11 blocks in that matrix.  And I have 6 sites.  So that matches 66, the number of rows that got added after the draft was saved.  It's at 2220 now after it saved another draft from the 2nd time I looked; another 66 rows.

[The 3.1.33 db was from last week, when we only had 5 blocks, I think, so that explains the 30 rows.  Additional blocks were authored-in a few hours ago.]

**Comment by ccsantos23** [comment]
After applying the 3.2.3 update, the fix for #4545 kept draft rows from being automatically added.  And when I deleted the couple drafts from yesterday, the rows for the matrixcontent_homebannermatrix table went back to 2088.  But a lot of those remaining rows must be automatic drafts generated when I first upgraded to 3.2.  Will there be a utility for cleaning those up?

**Comment by brandonkelly** [comment]
The only way a draft would have been automatically created is if you had actually gone to the entry’s edit page first (and were suffering from [this bug](https://github.com/craftcms/cms/issues/4545). Unfortunately there’s no difference between an auto-created draft and one that someone intentionally created, so no way to purge the auto-created drafts.

**Comment by ccsantos23** [comment]
I was looking for a way to remove **all** drafts.  Right now I can delete the drafts if  I can see them from the UI, but that example I gave was from a single (the home page), and I couldn't see where the drafts are to delete them.  It's like they're possibly orphaned in there.
Like #4564, I saw a bunch of "Updating entry drafts and revisions" when I upgraded to 3.2.

**Comment by brandonkelly** [comment]
If a draft isn’t showing up in the UI, my guess is that it doesn’t actually exist. Maybe you’re looking at revision data, not draft?

**Comment by ccsantos23** [comment]
@brandonkelly  When I tried querying my matrix's table against the elements table, I've found many rows created on 7/11/2019 (that was the date of the 3.2 upgrade), and they're showing draftId=NULL and revisionId=NULL.  Let me know if there's a different query I should try.  Thanks.

[query_ matrixcontent_homebannermatrix.pdf](https://github.com/craftcms/cms/files/3403054/query_.matrixcontent_homebannermatrix.pdf)

**Comment by brandonkelly** [comment]
There is no actual issue here. In Craft 3.2, entry drafts and revisions are stored as actual elements, whereas before they were stored as JSON blobs. Those elements will have actual duplicated Matrix block elements associated with them as well. So it’s expected that the size of `elements`, `matrixblocks` and other tables will grow in 3.2.

**Comment by ccsantos23** [comment]
@brandonkelly  Thanks.  I've already gone ahead and set maxRevisions to 10 to keep it down (since we can't see beyond 10 in the UI anyway).  
One last question--
Just to see, I tried this on a copy of my website:  turned off versioning on the section that had the home page (a Single) so that all the revisions got removed.  Looks like matrixblocks rows went down, but the matrixcontent_homebannermatrix stayed in the 2k+ range.  There are 11 blocks in the matrix, propagated to 6 sites.  No revisions and no drafts.  Shouldn't the matrixcontent_homebannermatrix just have something like 66 rows?
